crates/bootstrap_srv/src/http.rs

```diff
@@ -132,17 +138,14 @@ fn tokio_thread(
 
             let receiver = HttpReceiver(h_recv);
 
-            let (s_shutdown, r_shutdown) =
-                tokio::sync::oneshot::channel::<()>();
-            let r_shutdown = futures::future::FutureExt::shared(async move {
-                let _ = r_shutdown.await;
-            });
-
             let mut addrs = Vec::with_capacity(config.addrs.len());
-            let mut servers = Vec::with_capacity(config.addrs.len());
+            let mut servers: Vec<BoxFut<'static, Result<()>>> =
+                Vec::with_capacity(config.addrs.len());
+
+            let shutdown_handle = axum_server::Handle::new();
 
             for addr in config.addrs {
-                let listener = match tokio::net::TcpListener::bind(addr).await {
+                let listener = match std::net::TcpListener::bind(addr) {
```

Comment:
This is a blocking call in an async context.
- can just use `axum_server::Server::bind` instead
- or wrap this in a `spawn_blocking`
- or use [tokio into_std](https://docs.rs/tokio/1.43.0/tokio/net/struct.TcpListener.html#method.into_std)

Reply:
I couldn't see a way to get the local address back from option 1, wrapped in `spawn_blocking`